src/libraries/Common/src/Interop/OSX/Interop.Libraries.cs

```diff
@@ -15,5 +15,6 @@ internal static partial class Libraries
         internal const string SystemConfigurationLibrary = "/System/Library/Frameworks/SystemConfiguration.framework/SystemConfiguration";
         internal const string AppleCryptoNative = "libSystem.Security.Cryptography.Native.Apple";
         internal const string MsQuic = "libmsquic.dylib";
+        internal const string libc = "libc";
```

Comment:
As per the [interop guidelines](https://github.com/dotnet/runtime/blob/462ab6e7a65dec97944e408b1959e71276f83988/docs/coding-guidelines/interop-guidelines.md#submitting-changes), we should update https://github.com/dotnet/core `runtime-deps.json` for 7.0 release.
cc @dotnet/platform-deps-team.

Reply:
The c library is already a dependency.

Reply:
@tmds, I don't see `libc` as a dependency for macOS, there are other c libraries that are, but this is a different one.

I'm not sure what `runtime-deps.json` does, but if it copies the natives to the output `.app` maybe it shouldn't be copied for this one since it's possibly quite tied to the OS version? (I'm not sure). And it's guaranteed to be somewhere on macOS as well since many other (likely including system) apps would use it too. But I'm not an expert on this so take that with a grain of salt.

Reply:
`libc` is the standard C library. The OS provides it.

Reply:
Yes it does, but it is still not a dependency of .NET currently (https://github.com/dotnet/core/blob/main/release-notes/6.0/runtime-deps.json). Eg. `libobjc` is listed in that file which is always provided by macOS, so why shouldn't `libc` also be listed? (I'm not sure what the file does, but I am aware the OS provides the library always.)

Reply:
Since it's an inherent component of the OS, it's not listed.  The runtime-deps.json file is intended to be the source of truth for .NET's platform dependencies.  Consuming the contents of this file is a work-in-progress but you can find out more about it at https://github.com/dotnet/core/issues/5651.

Reply:
> Since it's an inherent component of the OS, it's not listed.

Okay, that makes sense. I thought that initially, but if that's the case then why is `libobjc.dylib` listed since it's an inherent component of the OS. Everything else on the osx list seems to be a custom depencency, but don't apple OSes come with `libobjc.dylib`, because otherwise how else would any objective c apps run (including apple's own), they all compile down to those apis don't they? Please correct me if I'm wrong.